test/Microsoft.TestPlatform.TestUtilities/IntegrationTestBase.cs

```diff
@@ -752,25 +766,7 @@ protected static string GetDiagArg(string rootDir)
     /// Counts the number of logs following the '*.host.*' pattern in the given folder.
     /// </summary>
     protected static int CountTestHostLogs(string diagLogsDir, IEnumerable<string> testHostProcessNames)
-        => Directory.GetFiles(diagLogsDir, "*.host.*")
-            .Count(filePath =>
-            {
-                var firstLine = File.ReadLines(filePath).FirstOrDefault();
-                return testHostProcessNames.Any(processName =>
-                {
-                    var parts = processName.Split('.');
-                    if (parts.Length > 2)
-                    {
-                        throw new InvalidOperationException("");
-                    }
-
-                    var hostName = parts[0];
-                    var platformName = parts.Length > 1 ? @$"\.{parts[1]}" : string.Empty;
-
-                    var isMatch = Regex.IsMatch(firstLine, @$",\s{hostName}(?:\.net\d+)?{platformName}\.(?:exe|dll),");
-                    return isMatch;
-                });
-            });
+        => Directory.GetFiles(diagLogsDir, "*.host.*").Count();
```

Comment:
[minor] `GetFiles` returns an array so there is no need to iterate to have the count.

```suggestion
        => Directory.GetFiles(diagLogsDir, "*.host.*").Length;
```

Reply:
I remember you mentioned something like that few times, but I don't think there is any enumeration going on. Array<T> is ICollection<T> so it will just return .Count. 

https://github.com/microsoft/referencesource/blob/master/System.Core/System/Linq/Enumerable.cs#L1317